File: tests/parser_integration/fixtures/javascript/string_format_preservation.js
// Test file for string format preservation
// This file tests various JavaScript string literal types

function testRegularStrings() {
    const s1 = "Hello, world!";
    const s2 = 'Simple text';
    console.log(s1);
}

function testTemplateStrings() {
    const name = "Alice";
    const formatted = `Hello, ${name}!`;
    console.log(formatted);
}

function testMultilineStrings() {
    const multiline = `Line 1
Line 2
Line 3`;
    console.log(multiline);
}

function testErrorMessages() {
    throw new Error("Something went wrong");
    console.error("Failed to connect");
}

function testEscapedStrings() {
    const escaped = "Hello\nWorld\t!";
    const quoted = 'He said "Hello"';
    console.log(escaped);
}

module.exports = {
    testRegularStrings,
    testTemplateStrings
};

File: tests/parser_integration/fixtures/rust/string_format_preservation.rs
// Test file for string format preservation
// This file tests various Rust string literal types

fn test_regular_strings() {
    let s1 = "Hello, world!";
    let s2 = "Simple text";
    println!("{}", s1);
}

fn test_raw_strings() {
    let raw1 = r"Hello, world!";
    let raw2 = r#"Hello, "world"!"#;
    let raw3 = r##"Hello, #"world"#!"##;
    println!("{}", raw1);
}

fn test_byte_strings() {
    let bytes = b"Hello, world!";
    println!("{:?}", bytes);
}

fn test_format_strings() {
    let name = "Alice";
    let formatted = format!("Hello, {}!", name);
    println!("{}", formatted);
}

fn test_error_messages() {
    panic!("Something went wrong");
    log::error!("Failed to connect");
}

fn test_multiline_strings() {
    let multiline = "Line 1\nLine 2\nLine 3";
    println!("{}", multiline);
}

fn test_strings_with_escapes() {
    let escaped = "Hello\nWorld\t!";
    let quoted = "He said \"Hello\"";
    println!("{}", escaped);
}
